Sources/LogInOut.php

```diff
@@ -809,9 +805,6 @@ function validatePasswordFlood($id_member, $member_name, $password_flood_value =
 	// Only if they're not validating for 2FA
 	if (!$tfa)
 	{
-		require_once($sourcedir . '/Subs-Auth.php');
-		setLoginCookie(-3600, 0);
```

Comment:
It's the key issue from #4297.
Because of this line of code we generate a new php session while we didn't call the checksession.
Small explain:
Login form with form_session id 123 -> send -> validatePasswordFlood -> change php_session id to 567 -> checksession call -> check form_session id(123) and php_session id(567) -> error "sesion time out.." 
so this issue: https://github.com/SimpleMachines/SMF2.1/issues/4365
get fixed by this and is reason why we get session timeout.

Reply:
I'm not sure that it is the key issue. Here's the code from the 2.0.x branch:

```
function validatePasswordFlood($id_member, $password_flood_value = false, $was_correct = false)
{
	global $smcFunc, $cookiename, $sourcedir;

	// As this is only brute protection, we allow 5 attempts every 10 seconds.

	// Destroy any session or cookie data about this member, as they validated wrong.
	require_once($sourcedir . '/Subs-Auth.php');
	setLoginCookie(-3600, 0);

	if (isset($_SESSION['login_' . $cookiename]))
		unset($_SESSION['login_' . $cookiename]);

```

This code doesn't lead to such problems in 2.0.x, and the relevant logic for how we might reach this point is unchanged in 2.1. 

Removing the code to reset the cookie from `validatePasswordFlood()` is a problem. If they user supplied the wrong login information, their cookie needs to expire right away.

Reply:
validatePasswordFlood() is called before checksession(),
so you had two possibility:
- call it later
- don't change the session

it choose last one

The issue in my test case only exists when you got a smfcookie which is not valid.
what i do to generate the issue:
Login in my pg env, 
change the setting to my mysql env,
try to login into my mysql env.
important: got the same cookie name

And sure exists the issue in 2.0.x also.

Reply:
I just tried it on 2.0.x, and everything behaved as it should.

Reply:
Okay maybe different  (I don't care 2.0.x).
For smf 2.1 it's work this way.

Reply:
So i checked smf 2.0.x this doesn't call validatePasswordFlood() in the login process.

Reply:
What is the idea by removing here the LoginCookie?